### PR TITLE
feat: Add Js API

### PR DESCRIPTION
**Pull request by enricocolasante**

### CODE REVIEW

File: src/jsMain/kotlin/org/hisp/dhis/rules/Logger.kt [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package org.hisp.dhis.rules
 
+import org.hisp.dhis.lib.expression.js.ExpressionJs
```
Comment (jbee) [inline src/jsMain/kotlin/org/hisp/dhis/rules/Logger.kt]: ? should be unused

File: src/jsMain/kotlin/org/hisp/dhis/rules/RuleDataValueJs.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package org.hisp.dhis.rules
+
+import kotlinx.datetime.internal.JSJoda.Instant
+
+@JsExport
+@OptIn(ExperimentalJsExport::class)
+data class RuleDataValueJs(
+    val eventDate: Instant,
```
Comment (jbee) [inline src/jsMain/kotlin/org/hisp/dhis/rules/RuleDataValueJs.kt:8]: I assume this means `Instant` can be used

**Review by jbee — APPROVED** [review]
You made quick work out of it :)

File: src/jsMain/kotlin/org/hisp/dhis/rules/RuleDataValueJs.kt (hunk 2 above)
Comment (vgarciabnz) [reply src/jsMain/kotlin/org/hisp/dhis/rules/RuleDataValueJs.kt:8]: That's interesting, it comes from `JSJoda.Instant`

File: src/jsMain/kotlin/org/hisp/dhis/rules/RuleEventJs.kt [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package org.hisp.dhis.rules
+
+import kotlinx.datetime.internal.JSJoda.Instant
+import kotlinx.datetime.internal.JSJoda.LocalDate
```
Comment (vgarciabnz) [inline src/jsMain/kotlin/org/hisp/dhis/rules/RuleEventJs.kt:4]: Nice, I didn't know about it

File: src/jsMain/kotlin/org/hisp/dhis/rules/RuleVariableJs.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    val programStage: String?
+){
+    init {
+        require(RULE_VALUE_TYPES.contains(fieldType)) { "RuleValueType type must be one of: $RULE_VALUE_TYPES" }
+        require(TYPES.contains(fieldType)) { "RuleVariableType type must be one of: $TYPES" }
+    }
+
+    companion object {
+        val RULE_VALUE_TYPES = RuleValueType.entries.map { it.name } .toTypedArray()
+        val TYPES = arrayOf("DATAELEMENT_NEWEST_EVENT_PROGRAM_STAGE",
```
Comment (vgarciabnz) [inline src/jsMain/kotlin/org/hisp/dhis/rules/RuleVariableJs.kt]: This list of strings, couldn't be replaced by `RuleVariableType.entries.map { it.name } .toTypedArray()`? The same way you did for RULE_VALUE_TYPES

File: src/jsMain/kotlin/org/hisp/dhis/rules/RuleDataValueJs.kt (hunk 2 above)
Comment (enricocolasante) [reply src/jsMain/kotlin/org/hisp/dhis/rules/RuleDataValueJs.kt:8]: Let's see how this works for frontenders

**Review by vgarciabnz — APPROVED** [review]

**Pull request merged**